File: lab 7/scala/src/main/scala/Main.scala
import java.util

import scala.collection.mutable.ListBuffer


object Main {
  def main(args: Array[String]): Unit =
  {
    val n = 7
        val philosophers = new ListBuffer[Philosopher]
        val forks = new ListBuffer[Fork]
        for {i <- 0 to n} forks += (new Fork)
        for (i <- 0 to n)philosophers += (new AsymetricPhilosopher(i, forks(i % n), forks((i + 1) % n)))
        val simulation = new Simulation(philosophers)
        val result = simulation.simulate
        System.out.println(result)


  }
}

File: lab 7/java/src/Main.java
import java.util.ArrayList;
        import java.util.LinkedList;

public class Main
{
    public static void main(String[] args) {

        for(int N = 2; N < 10; N++)
        {
            LinkedList<Philosopher> philosophers = new LinkedList<>();
            ArrayList<Fork> forks = new ArrayList<>();

            for(int i = 0; i < N; i++)
            {
                forks.add(new Fork());
            }

            Table table = new Table(N-1);
            for(int i = 0; i < N; i++)
            {
                philosophers.add(new NaivePhilosopher(i, forks.get(i%N), forks.get((i+1)%N)));
            }

            Simulation simulation = new Simulation(philosophers);
            String result = simulation.simulate();
            System.out.println(result);
        }
    }
}
